test/contraction/test_contraction.cpp

```diff
@@ -122,17 +130,31 @@ TYPED_TEST_SUITE(TestContractionScale, ScaleKernelTypes);
 TYPED_TEST(TestContractionBilinear, bilinear)
 {
     this->p_cd_element_op = std::make_unique<Bilinear>(1.f, 1.f);
-    this->Run();
+    this->template Run<6>({{2, 3, 2, 3, 2, 3}, {2, 3, 2, 3, 2, 3}, {2, 2, 2, 2, 2, 4}});
+    this->template Run<6>({{1, 1, 1, 3, 2, 3}, {1, 1, 1, 3, 2, 3}, {1, 1, 1, 2, 2, 4}});
+    this->template Run<2>({{16, 16}, {16, 16}, {16, 16}});
+    this->template Run<2>({{8, 8}, {16, 16}, {8, 8}});
```

Comment:
Please use different values for all Ms, Ns, and Ks here and in all other test cases.

Reply:
Done.